Change default TF-M profile for bluetooth/openthread/mesh

Vge0rge opened the pull request:
See commit messages

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [875616a4a00e0aa5327585fe4f1c08eab52c86d0](https://github.com/nrfconnect/sdk-nrf/pull/24059/commits/875616a4a00e0aa5327585fe4f1c08eab52c86d0)
zephyr: PR head: [9de5e8062ce2b733c7b4bcb1e244f2f78a9ec55e](https://github.com/nrfconnect/sdk-zephyr/pull/3210/commits/9de5e8062ce2b733c7b4bcb1e244f2f78a9ec55e)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [875616a4a00e0aa5327585fe4f1c08eab52c86d0](https://github.com/nrfconnect/sdk-nrf/pull/24059/commits/875616a4a00e0aa5327585fe4f1c08eab52c86d0)
> merge base: [94521ce6cf243c8ae3e30d723768fa55d6e0c8db](https://github.com/nrfconnect/sdk-nrf/commit/94521ce6cf243c8ae3e30d723768fa55d6e0c8db)
> target head (main): [94521ce6cf243c8ae3e30d723768fa55d6e0c8db](https://github.com/nrfconnect/sdk-nrf/commit/94521ce6cf243c8ae3e30d723768fa55d6e0c8db)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/94521ce6cf243c8ae3e30d723768fa55d6e0c8db..875616a4a00e0aa5327585fe4f1c08eab52c86d0)

zephyr:
> PR head: [9de5e8062ce2b733c7b4bcb1e244f2f78a9ec55e](https://github.com/nrfconnect/sdk-zephyr/pull/3210/commits/9de5e8062ce2b733c7b4bcb1e244f2f78a9ec55e)
> merge base: [f637c2ca51eee3c26bcc6a53619f4dd1c64d3b4f](https://github.com/nrfconnect/sdk-zephyr/commit/f637c2ca51eee3c26bcc6a53619f4dd1c64d3b4f)
> target head (main): [1fe49abe2fccbdacdfbc9b2876d6ca0245963085](https://github.com/nrfconnect/sdk-zephyr/commit/1fe49abe2fccbdacdfbc9b2876d6ca0245963085)
> [Diff](https://github.com/nrfconnect/sdk-zephyr/compare/f637c2ca51eee3c26bcc6a53619f4dd1c64d3b4f..9de5e8062ce2b733c7b4bcb1e244f2f78a9ec55e)
</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (7)</summary>

```
modules
│  ├── trusted-firmware-m
│  │  │ Kconfig.tfm.defconfig
subsys
│  ├── bluetooth
│  │  ├── mesh
│  │  │  │ Kconfig
│  ├── net
│  │  ├── openthread
│  │  │  │ Kconfig
tests
│  ├── subsys
│  │  ├── bluetooth
│  │  │  ├── fast_pair
│  │  │  │  ├── crypto
│  │  │  │  │  │ Kconfig
west.yml
zephyr
│  ├── tests
│  │  ├── application_development
│  │  │  ├── vector_table_relocation
│  │  │  │  ├── src
│  │  │  │  │  │ main.c
│  │  ├── bluetooth
│  │  │  ├── bt_crypto
│  │  │  │  ├── boards
│  │  │  │  │  │ nrf5340dk_nrf5340_cpuapp_ns.conf
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests
   - :orange_circle: test-fw-nrfconnect-ble_mesh
   - :orange_circle: test-fw-nrfconnect-chip
   - :orange_circle: test-fw-nrfconnect-nrf_crypto
   - :orange_circle: test-fw-nrfconnect-tfm
   - :orange_circle: test-fw-nrfconnect-thread-main
   - :orange_circle: test-sdk-find-my
   - :orange_circle: test-sdk-mcuboot
   - :orange_circle: test-sdk-dfu

<details><summary>Disabled integration tests</summary>

- 
   - test-fw-nrfconnect-nrf_lrcs_mosh
   - test-fw-nrfconnect-nrf_lrcs_positioning
   - desktop52_verification
   - doc-internal
   - test_ble_nrf_config
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_samples
   - test-fw-nrfconnect-fem
   - test-fw-nrfconnect-nfc
   - test-fw-nrfconnect-nrf-iot_cloud
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - test-fw-nrfconnect-nrf-iot_thingy91
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-proprietary_esb
   - test-fw-nrfconnect-ps-main
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-rs
   - test-low-level
   - test-sdk-audio
   - test-sdk-pmic-samples
   - test-sdk-wifi
   - test-secdom-samples-public
</details>



*Note: This message is automatically posted and updated by the CI*

alxelax reviewed (commented):
From mesh point of view the changes are Ok. 
However, since dependency on TEST... macro was removed than changes from downstream shouldn't be reverted. Right?

Vge0rge commented:
> From mesh point of view the changes are Ok. However, since dependency on TEST_BT_FAST_PAIR_CRYPTO_PSA macro was removed, then changes from downstream shouldn't be reverted. Right?

@alxelax I am not sure that I follow, I removed the dependency to the TEST_BT_FAST_PAIR_CRYPTO_PSA because of CI issues and reverted my original change to the Kconfig file which enabled the correct TF-M profile for this test. Am I missing something here?

alxelax commented:
> > From mesh point of view the changes are Ok. However, since dependency on TEST_BT_FAST_PAIR_CRYPTO_PSA macro was removed, then changes from downstream shouldn't be reverted. Right?
> 
> @alxelax I am not sure that I follow, I removed the dependency to the TEST_BT_FAST_PAIR_CRYPTO_PSA because of CI issues and reverted my original change to the Kconfig file which enabled the correct TF-M profile for this test. Am I missing something here?

Probably I missed something. @PavelVPV, could you confirm that after reverting your fix from downstream(https://github.com/nrfconnect/sdk-zephyr/commit/319b359ae7669cd64ba3adec65034a4a4e4a5535) and these changes the test(ble crypto test that you fixed) still passes?

alxelax commented:
> > > > From mesh point of view the changes are Ok. However, since dependency on TEST_BT_FAST_PAIR_CRYPTO_PSA macro was removed, then changes from downstream shouldn't be reverted. Right?
> > > 
> > > 
> > > @alxelax I am not sure that I follow, I removed the dependency to the TEST_BT_FAST_PAIR_CRYPTO_PSA because of CI issues and reverted my original change to the Kconfig file which enabled the correct TF-M profile for this test. Am I missing something here?
> > 
> > 
> > Probably I missed something. @PavelVPV, could you confirm that after reverting your fix from downstream([nrfconnect/sdk-zephyr@319b359](https://github.com/nrfconnect/sdk-zephyr/commit/319b359ae7669cd64ba3adec65034a4a4e4a5535)) and applying of these changes the test(ble crypto test that you fixed) still passes?
> 
> This revert was discussed internally. It was passing locally on @Vge0rge machine, and we discussed how to run CI. But if needed I can double check locally.

Not needed. If you run and passed it, then this is Ok. I'll approve PR after it passes CI steps.

Vge0rge commented:
> @Vge0rge , do you plan to backport this to 3.1.1?

I didn't have that in mind, no.

grochu commented:
@Vge0rge, please add relevant notes in the NCS changelog in documentation, so the user have it clear we will no longer support them.

NordicBuilder commented:
### Memory footprint analysis revealed the following potential issues

***sample.find_my.switchable_networks.release.ui_switch[nrf54lm20dk/nrf54lm20a/cpuapp]***: ***ROM*** size  ***increased by 4080[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=find_my&var-scenario=switchable_networks&var-testcase=release.ui_switch&var-arch=arm&var-platform=nrf54lm20dk/nrf54lm20a/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***sample.matter.template.debug[nrf5340dk/nrf5340/cpuapp]***: ***ROM*** size  ***increased by 2156[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=debug&var-arch=arm&var-platform=nrf5340dk/nrf5340/cpuapp) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic)
***sample.matter.template.release[nrf5340dk/nrf5340/cpuapp]***: ***ROM*** size  ***increased by 1596[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=release&var-arch=arm&var-platform=nrf5340dk/nrf5340/cpuapp) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic)
***sample.matter.template.release[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1648[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=release&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic)
***sample.find_my.locator_tag.release[nrf54lm20dk/nrf54lm20a/cpuapp]***: ***ROM*** size  ***increased by 2520[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=find_my&var-scenario=locator_tag&var-testcase=release&var-arch=arm&var-platform=nrf54lm20dk/nrf54lm20a/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***sample.matter.template.debug[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 2156[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=matter&var-scenario=template&var-testcase=debug&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @kkasperczyk-no @ArekBalysNordic @markaj-nordic)
***sample.bluetooth.fast_pair.locator_tag.release[nrf54lm20dk/nrf54lm20a/cpuapp]***: ***ROM*** size  ***increased by 4024[B]*** in comparison to the ***main[94521ce6]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=sample&var-submodule=bluetooth&var-scenario=fast_pair&var-testcase=locator_tag.release&var-arch=arm&var-platform=nrf54lm20dk/nrf54lm20a/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)


*Note: This message is automatically posted and updated by the CI (latest/sdk-nrf/PR-24059/7)*

anangl commented on west.yml:
```diff
@@ -65,7 +65,7 @@ manifest:
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: 59e68646df7213ce3e782f781fdb872cfdc42f7d
+      revision: pull/3210/head
```
Update with SHA: `dc15a79bfeb842bfb503fd63017e8cb371050ddf`.
